Unit tests for getFilteredBillingTransactions fail in certain timezones

To reproduce:
1. Set the system timezone to New York, GMT-4
2. `npm run test-client client/state/selectors/test/get-filtered-billing-transactions.js`
3. Observe the failure
4. Set the system timezone to London, BST
5. Re-run the test
6. The test passes 

CC @blowery @jsnajdr

I tested and confirmed that running `npm run test-client client/state/selectors/test/get-filtered-billing-transactions.js` results in all tests passing when my system timezone is set to London GMT-0 and that I see the following errors for the same tests with my system timezone set to Denver, GMT-6.

```
Test Suites: 1 failed, 1 total
Tests:       1 failed, 12 passed, 13 total
Snapshots:   0 total
Time:        2.461s
Ran all test suites matching /client\/state\/selectors\/test\/get-filtered-billing-transactions.js/i.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! wp-calypso@0.17.0 test-client: `jest -c=test/client/jest.config.json "client/state/selectors/test/get-filtered-billing-transactions.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the wp-calypso@0.17.0 test-client script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/designsimply/.npm/_logs/2018-05-18T15_47_32_927Z-debug.log
```

--- a/client/state/selectors/test/get-filtered-billing-transactions.js
+++ b/client/state/selectors/test/get-filtered-billing-transactions.js
@@ -19,7 +19,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 			items: {
 				past: [
 					{
-						date: '2018-05-01T12:00:00+0000',
+						date: '2018-05-01T12:00:00',
 						service: 'WordPress.com',
 						cc_name: 'name1 surname1',
 						cc_type: 'mastercard',
@@ -32,7 +32,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2018-04-11T13:11:27+0000',
+						date: '2018-04-11T13:11:27',
 						service: 'WordPress.com',
 						cc_name: 'name2',
 						cc_type: 'visa',
@@ -50,7 +50,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2018-03-11T21:00:00+0000',
+						date: '2018-03-11T21:00:00',
 						service: 'Store Services',
 						cc_name: 'name1 surname1',
 						cc_type: 'visa',
@@ -68,7 +68,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2018-03-15T10:39:27+0000',
+						date: '2018-03-15T10:39:27',
 						service: 'Store Services',
 						cc_name: 'name2',
 						cc_type: 'mastercard',
@@ -86,7 +86,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2018-03-13T16:10:45+0000',
+						date: '2018-03-13T16:10:45',
 						service: 'WordPress.com',
 						cc_name: 'name1 surname1',
 						cc_type: 'visa',
@@ -99,7 +99,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2018-01-10T14:24:38+0000',
+						date: '2018-01-10T14:24:38',
 						service: 'WordPress.com',
 						cc_name: 'name2',
 						cc_type: 'mastercard',
@@ -112,7 +112,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2017-12-10T10:30:38+0000',
+						date: '2017-12-10T10:30:38',
 						service: 'Store Services',
 						cc_name: 'name1 surname1',
 						cc_type: 'visa',
@@ -125,7 +125,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2017-12-01T07:20:00+0000',
+						date: '2017-12-01T07:20:00',
 						service: 'Store Services',
 						cc_name: 'name1 surname1',
 						cc_type: 'visa',
@@ -138,7 +138,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2017-11-24T05:13:00+0000',
+						date: '2017-11-24T05:13:00',
 						service: 'WordPress.com',
 						cc_name: 'name1 surname1',
 						cc_type: 'visa',
@@ -151,7 +151,7 @@ describe( 'getBillingTransactionAppFilterValues()', () => {
 						],
 					},
 					{
-						date: '2017-01-01T00:00:00+0000',
+						date: '2017-01-01T00:00:00',
 						service: 'Store Services',
 						cc_name: 'name2',
 						cc_type: 'mastercard',

